cmd/function-sidecar.go

```diff
@@ -80,27 +79,19 @@ func main() {
 		select {
 		case msg, ok := <-consumer.Messages():
 			if ok {
-				messageIn, err := message.ExtractMessage(msg.Value)
-				fmt.Fprintf(os.Stdout, "<<< %s\n", messageIn)
-				if err != nil {
-					log.Printf("Error receiving message from Kafka: %v", err)
-					break
-				}
-				strPayload := string(messageIn.Payload.([]byte))
+				strPayload := string(msg.Value);
 				dispatched, err := dispatcher.Dispatch(strPayload)
 				if err != nil {
 					log.Printf("Error dispatching message: %v", err)
 					break
 				}
 				if output != nil {
-					messageOut := message.Message{Payload: []byte(dispatched.(string)), Headers: messageIn.Headers}
-					bytesOut, err := message.EncodeMessage(messageOut)
-					fmt.Fprintf(os.Stdout, ">>> %s\n", messageOut)
-					if err != nil {
-						log.Printf("Error encoding message: %v", err)
-						break
+					bytesOut := []byte(dispatched.(string));
+					replyHeaders := make([]sarama.RecordHeader, len(msg.Headers))
+					for i, h := range msg.Headers {
```

Comment:
Should be able to use the builtin `copy`, if not pass the incoming slice directly (I assume you wanted to make a semantic distinction between the two?)

Reply:
I agree there should be a simpler way, but after trying several things, this was all I got to work for now. That said, it's clearly just my misunderstanding of how dereferencing works with arrays/slices at this point (including when it's handled automatically). Here's what I tried first (passing the inbound headers as-is):

```
outMessage := &sarama.ProducerMessage{Topic: output.(string), Value: sarama.ByteEncoder(bytesOut), Headers: msg.Headers}
```

And that led to this error from `make dockerize`:

```
cmd/function-sidecar.go:94:112: cannot use msg.Headers (type []*sarama.RecordHeader) as type []sarama.RecordHeader in field value
```

I also tried dereferencing at the slice level, but that led to syntax errors.

Then I tried `copy(replyHeaders, msg.Headers)`, but received a similar build error:

```
cmd/function-sidecar.go:91:10: arguments to copy have different element types: []sarama.RecordHeader and []*sarama.RecordHeader
```

Again, I experimented with dereferencing syntax and a couple different ways of declaring `replyHeaders` but I ended up with the range and item-by-item pointer copy as something that worked. I'd be happy to learn how to handle this as closely as possible to the direct pass-thru I tried above (working on that learning part now, but feel free to share some knowledge ;-)).

Reply:
Ah nevermind, I did not even see that there was de-referencing involved. golang auto accepting pointers for values is still a bit too magical to me.